timesketch/api/v1/resources/sigma.py

```diff
@@ -47,53 +58,201 @@ def get(self):
+                            "parsed_rule":parsed_rule,
+                        }
+                    )
 
         except ValueError as e:
             logger.error(
-                "OS Error, unable to get the path to the Sigma rules", exc_info=True
+                "Error, unable to get Sigma rules", exc_info=True
             )
             abort(HTTP_STATUS_CODE_NOT_FOUND, f"Value Error, {e}")
-        # TODO: idea for meta: add a list of folders that have been parsed
         meta = {"current_user": current_user.username, "rules_count": len(sigma_rules)}
         return jsonify({"objects": sigma_rules, "meta": meta})
 
 
-class SigmaResource(resources.ResourceMixin, Resource):
+class SigmaRuleResource(resources.ResourceMixin, Resource):
     """Resource to get a Sigma rule."""
 
     @login_required
     def get(self, rule_uuid):
         """Handles GET request to the resource.
 
         Args:
-            rule_uuid: uuid of the sigma rule
+            rule_uuid: uuid of the rule
 
         Returns:
             JSON sigma rule
         """
-        return_rule = None
         try:
-            sigma_rules = ts_sigma_lib.get_all_sigma_rules()
+            rule = SigmaRule.query.filter_by(rule_uuid=rule_uuid).first()
 
-        except ValueError as e:
+        except Exception as e: # pylint: disable=broad-except
             logger.error(
-                "OS Error, unable to get the path to the Sigma rules", exc_info=True
+                "Unable to get the Sigma rule",
+                exc_info=True,
             )
             abort(HTTP_STATUS_CODE_NOT_FOUND, f"ValueError {e}")
-        for rule in sigma_rules:
-            if rule is not None:
-                if rule_uuid == rule.get("id"):
-                    return_rule = rule
 
-        if return_rule is None:
-            abort(HTTP_STATUS_CODE_NOT_FOUND, "No sigma rule found with this ID.")
+        if not rule:
+            abort(HTTP_STATUS_CODE_NOT_FOUND, "No rule found with this ID.")
+        return_rules = []
+
+        assert isinstance(rule, SigmaRule)
+        # Return a subset of the sigma objects to reduce the amount of
+        # data sent to the client.
+
+        parsed_rule = ts_sigma_lib.get_sigma_rule_by_text(rule.rule_yaml)
+        # check if the rule_uuid that is parsed out matches the stored id
+        if rule_uuid == parsed_rule.pop("id"):
+            parsed_rule["rule_uuid"] = parsed_rule.pop("id")
+        else:
+            parsed_rule["rule_uuid"] = rule_uuid
+        parsed_rule["id"] = rule.id #this is the database id (1,2,3...)
+        parsed_rule["created_at"] = str(rule.created_at)
+        parsed_rule["updated_at"] = str(rule.updated_at)
+        parsed_rule["status"] = rule.get_status.status
+        parsed_rule["title"] = rule.title
+        parsed_rule["description"] = rule.description
+        parsed_rule["rule_yaml"] = rule.rule_yaml
+        return_rules.append(
+            parsed_rule)
```

Comment:
Seeing we are doing this in two resources, I opt to create a private util function at the top of this file to do this

Reply:
refactored it and called it `_enrich_sigma_rule_object`WDYT?